models/project/github.go

```diff
@@ -0,0 +1,202 @@
+package project
+
+import (
+	"../../config"
+	"../../modules/hub"
+	"../../modules/logging"
+	"../db"
+	"../list"
+	"../task"
+	"database/sql"
+	"errors"
+
+	"github.com/google/go-github/github"
+)
+
+func (u *ProjectStruct) LoadFromGithub(issues []github.Issue) error {
+	for _, issue := range issues {
+		targetTask, _ := task.FindByIssueNumber(u.ID, *issue.Number)
+
+		err := u.taskApplyLabel(targetTask, &issue)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func IssuesEvent(repositoryID int64, body github.IssuesEvent) error {
+	objectDB := &db.Database{}
+	var interfaceDB db.DB = objectDB
+	table := interfaceDB.Init()
+	defer table.Close()
+
+	var projectID int64
+	err := table.QueryRow("select id from projects where repository_id = ?;", repositoryID).Scan(&projectID)
+	if err != nil {
+		return err
+	}
+	parentProject := FindProject(projectID)
+	targetTask, _ := task.FindByIssueNumber(projectID, *body.Issue.Number)
+
+	switch *body.Action {
+	case "opened", "reopened":
+		if targetTask == nil {
+			err = parentProject.createNewTask(body.Issue)
+		} else {
+			err = parentProject.reopenTask(targetTask, body.Issue)
+		}
+	case "closed", "labeled", "unlabeled":
+		err = parentProject.taskApplyLabel(targetTask, body.Issue)
+	}
+	return err
+}
+
+func PullRequestEvent(repositoryID int64, body github.PullRequestEvent) error {
```

Comment:
__[golint]__
* [golint] exported function PullRequestEvent should have comment or be unexported